### PR TITLE
BinaryOpConditionsCollector: Add tests and fix edge case

### PR DESCRIPTION
**Pull request by jtojnar**

### CODE REVIEW

File: packages/NodeCollector/BinaryOpConditionsCollector.php [hunk 1]
```diff
@@ -23,6 +23,10 @@ final class BinaryOpConditionsCollector
      */
     public function findConditions(BinaryOp $binaryOp, string $binaryOpClass): array
     {
+        if ($binaryOp::class !== $binaryOpClass) {
```
Comment (jtojnar) [inline packages/NodeCollector/BinaryOpConditionsCollector.php:26]: Alternately, I can make this an assertion but considering it a trivial case with zero operators feels more natural and also composes better with the next thing I got planned.

**Comment by TomasVotruba** [comment]
Thank you :+1: 

Can you add test case to responsible Rector rule, that failed before? So we have releated Rector change example covered.

**Comment by jtojnar** [comment]
This does not affect a rector rule yet. I need it for https://github.com/rectorphp/rector-src/pull/1608, where I will add a test.

**Comment by TomasVotruba** [comment]
Ok, thank you :+1:
